src/TreeHouse/BuckarooBundle/Report/SimpleSepaDirectDebitTransactionDebitReport.php

```diff
@@ -42,12 +31,18 @@ public static function create(array $data)
             'BRQ_TRANSACTIONS',
         ];
 
-        foreach ($requiredFields as $field) {
-            if (!isset($data[$field])) {
-                throw new \InvalidArgumentException(sprintf('Missing field: %s', $field));
-            }
+        static::checkRequiredFields($requiredFields, $data);
+
+        if ('C008' !== $data['BRQ_TRANSACTION_TYPE']) {
+            throw new \RuntimeException(
+                'Expected to create a %s for a debit transaction. Got a credit transaction (type %s) instead. ' .
+                'Are you attempting to create a SimpleSepaDirectDebitTransactionDebitReport?',
+                get_class(),
+                $data['BRQ_TRANSACTION_TYPE']
+            );
```

Comment:
This won't work; missing a `sprintf` call (this was an existing error, but may as well fix it while you're at it). 

The `get_class()` call can be replaced with `__CLASS__` if I read it correctly.